### PR TITLE
add a setup check for 64 bit

### PR DESCRIPTION
**Pull request by szaimen**
See https://github.com/nextcloud/documentation/pull/9071

Signed-off-by: szaimen <szaimen@e.mail.de>

### CODE REVIEW

File: core/js/tests/specs/setupchecksSpec.js [hunk 1]
```diff
@@ -1333,6 +1352,64 @@ describe('OC.SetupChecks tests', function() {
 			});
 		});
 
+		it('should return an error if gmp or bcmath are not enabled', function(done) {
```
Comment (mejo-) [inline core/js/tests/specs/setupchecksSpec.js:1355]: The test name should probably be adapted.

Comment (szaimen) [reply core/js/tests/specs/setupchecksSpec.js:1355]: fixed

**Comment by szaimen** [comment]
@MichaIng this check will only be present on NC25 and above and is not going to be backported.

**Review by MichaIng — APPROVED** [review]
Makes sense. We should link the related docs section as well then. I'll add a list of known issues with 32-bit systems there.

But keep in mind that this likely leads to user reports like "Why suddenly my system is not supported anymore?"/"How can I mute the warning?" 😉.

**Comment by szaimen** [comment]
> We should link the related docs section as well then.

good idea!

**Comment by MichaIng** [comment]
> That is why we will not backport this change

But users will upgrade their Nextcloud instance and then see the warning, and at least I'd be very unhappy if the upgrade would be denied on my little RPi 2 home server 😉. But yeah, major upgrade means breaking changes/support changes are possible.

**Comment by szaimen** [comment]
We are not actively blocking the upgrade. Only showing this warning then!

**Review by PVince81 — APPROVED** [review]
:+1:

**Comment by MichaIng** [comment]
> in theory the check could also be added in the updater in NC < 25 close to the update button so that people get a warning before clicking "update".

That could be irritating since nothing changed about code-wise support but only a new warning is shown, which shouldn't be a reason to not update.

Or is there a plan to stop end user support when a 32-bit system is used, also not related to the known issues with it, exactly with NC25, while keeping it with NC24?

The requirements update in docs has been backported, which answers this already.

**Comment by szaimen** [comment]
> in theory the check could also be added in the updater in NC < 25 close to the update button so that people get a warning before clicking "update". of course only applies for web updater, not CLI phar update or other ways

I would rather not do this. Having the setup check is enough, imo.

**Comment by szaimen** [comment]
I'll do the link to the docs in a follow-up

**Comment by szaimen** [comment]
CI failure unrelated

**Pull request merged**